extra_tests/snippets/set.py

```diff
@@ -195,16 +195,18 @@ def __hash__(self):
 i = iter(a)
 a.add(4)
 a.remove(4)
-assert next(i) == 1
+with assert_raises(RuntimeError):
+    next(i) == 1
 
 a = set([1, 2, 3])
 i = iter(a)
 a.add(4)
 with assert_raises(RuntimeError):
     next(i)
 a.remove(4)
-with assert_raises(RuntimeError):
-    next(i)
+# TODO: Raises RuntimeError in CPython but we currently raise Runtime error.
```

Comment:
Update: After checking CPython, seems like this continues raising `RuntimeError`. We currently have no easy way of encoding this information for dicts and sets, I'll look into it at a following PR.

Reply:
Do you need to work more on here?

Reply:
Yup, I'll try and match CPythons behavior here but I'm leaving it for another PR since the dict iterators also need to be tweaked (and the test for them needs to be added in `extra_tests`).